src/commands.py

```diff
@@ -578,3 +535,55 @@ async def go_to_commons(machine: Machine, interaction, action: Action):
     @classmethod
     def helptxt(cls):
         return "$menu [dining commons] [mealtime] \nDisplays an interactable Embed showing the menu for the mealtime of the dining commons."
+
+class help(command_on_message):
+    """
+    $help [command]
+
+    Displays description of provided command. If no command is provided, displays all commands.
+    If a command is mod-only, have its helptxt return mod_only_command
+    """
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        self.commands_list = {
+            cmd.__name__: cmd.helptxt()
+            for cmd in command_on_message.__subclasses__()
+        }
+
+    async def action(self, message):
+        fields = []
+        show_mod = message.channel.id in (498634483910574082, 338237628275097601)
+        try:
+            cmd = message.content[1:].split()[1] # Tries to get the command to search for
+        except IndexError:
+            for name, desc in self.commands_list.items():
+                if isinstance(desc, mod_only_command) and not show_mod:
+                    continue
+                fields.append({
+                    "name": name,
+                    "value": desc
+                })
+
+        else: # If a command is given
+            if cmd in self.commands_list:
+                desc = self.commands_list[cmd]
+                if isinstance(desc, mod_only_command) and not show_mod:
```

Comment:
We should avoid using `isinstance`. It's generally not a good idea to couple class names with functionality unless it provides a significant benefit, or it is impossible to work without. Maybe we should change how the mod_only_command determines the string to be printed.